packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -110,29 +147,66 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
         'Check https://developers.google.com/identity/protocols/googlescopes '
         'for a list of valid OAuth 2.0 scopes.');
 
+    _isInitCalled = true; // Mark `init` ASAP before going async.
+
     await _jsSdkLoadedFuture;
 
-    _gisClient = overrideClient ??
-        GisSdkClient(
-          clientId: appClientId!,
-          hostedDomain: params.hostedDomain,
-          initialScopes: List<String>.from(params.scopes),
-          loggingEnabled: kDebugMode,
-        );
+    _gisClient ??= GisSdkClient(
+      clientId: appClientId!,
+      hostedDomain: params.hostedDomain,
+      initialScopes: List<String>.from(params.scopes),
+      userDataController: _userDataController,
+      loggingEnabled: kDebugMode,
+    );
+
+    _initDone.complete(); // Signal that `init` is fully done.
+  }
+
+  // Register a factory for the Button HtmlElementView.
+  void _registerButtonFactory() {
+    // ignore: avoid_dynamic_calls, undefined_prefixed_name
+    ui.platformViewRegistry.registerViewFactory(
+      'gsi_login_button',
+      (int viewId) {
+        final DomElement element = createDomElement('div');
+        element.setAttribute('style',
+            'width: 100%; height: 100%; overflow: hidden; display: flex; flex-wrap: wrap; align-content: center; justify-content: center;');
+        element.id = 'sign_in_button_$viewId';
+        return element;
+      },
+    );
+  }
 
-    _isInitCalled = true;
+  /// Render the GSI button web experience.
+  Widget renderButton({GSIButtonConfiguration? configuration}) {
+    final GSIButtonConfiguration config =
+        configuration ?? GSIButtonConfiguration();
+    return FutureBuilder<void>(
+      key: Key(config.hashCode.toString()),
+      future: initialized,
+      builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (snapshot.hasData) {
+          return FlexHtmlElementView(
+              viewType: 'gsi_login_button',
+              onPlatformViewCreated: (int viewId) {
+                final DomElement? element =
+                    domDocument.querySelector('#sign_in_button_$viewId');
+                assert(element != null,
+                    'Cannot render GSI button. DOM is not ready!');
+                _gisClient!.renderButton(element!, config);
+              });
+        }
+        return const Text('Getting ready');
+      },
+    );
   }
 
   @override
   Future<GoogleSignInUserData?> signInSilently() async {
     await initialized;
 
-    // Since the new GIS SDK does *not* perform authorization at the same time as
-    // authentication (and every one of our users expects that), we need to tell
-    // the plugin that this failed regardless of the actual result.
-    //
-    // However, if this succeeds, we'll save a People API request later.
-    return _gisClient.signInSilently().then((_) => null);
+    // The new user is being injected from the `userDataEvents` Stream.
+    return _gisClient!.signInSilently(); //.then((_) => null);
```

Comment:
nit: dead code starting with `//.then(...`

Reply:
Good eye, removing these across the package.